src/generate/device.rs

```diff
@@ -300,48 +300,42 @@ pub fn render(d: &Device, config: &Config, device_x: &mut String) -> Result<Toke
         }
     }
 
-    let span = Span::call_site();
-    let take = match config.target {
-        Target::CortexM => Some(Ident::new("cortex_m", span)),
-        Target::Msp430 => Some(Ident::new("msp430", span)),
-        Target::RISCV => Some(Ident::new("riscv", span)),
-        Target::XtensaLX => Some(Ident::new("xtensa_lx", span)),
-        Target::Mips => Some(Ident::new("mips_mcu", span)),
-        Target::None => None,
-    }
-    .map(|krate| {
-        quote! {
-            ///Returns all the peripherals *once*
-            #[inline]
-            pub fn take() -> Option<Self> {
-                #krate::interrupt::free(|_| {
-                    if unsafe { DEVICE_PERIPHERALS } {
-                        None
-                    } else {
-                        Some(unsafe { Peripherals::steal() })
-                    }
-                })
-            }
-        }
-    });
-
     out.extend(quote! {
         // NOTE `no_mangle` is used here to prevent linking different minor versions of the device
         // crate as that would let you `take` the device peripherals more than once (one per minor
         // version)
         #[no_mangle]
         static mut DEVICE_PERIPHERALS: bool = false;
 
-        ///All the peripherals
+        /// All the peripherals.
         #[allow(non_snake_case)]
         pub struct Peripherals {
             #fields
         }
 
         impl Peripherals {
-            #take
+            /// Returns all the peripherals *once*.
+            #[cfg(feature = "critical-section")]
```

Comment:
This means a new dependency, the crate documentation should be adjusted to account for this.


also, could https://github.com/rust-embedded/svd2rust/blob/a39a61d8ac8928a59d8a2b855c3303a766fd0517/ci/svd2rust-regress/src/svd_test.rs#L8 be updated to include `critical-section`